src/CodeGen_PTX_Dev.cpp

```diff
@@ -791,6 +809,255 @@ bool CodeGen_PTX_Dev::supports_atomic_add(const Type &t) const {
 
 }  // namespace
 
+class IsMatrixMultiply : public IRVisitor {
+    using IRVisitor::visit;
+
+public:
+    bool matmul_found = false;
+
+    Expr A;
+    Expr B;
+    Expr C;
+
+    std::string k_var_name;
+
+    IsMatrixMultiply(std::string k_var)
+        : k_var_name{std::move(k_var)} {
+    }
+
+    void visit(const Store *store) override {
+        //  matmul$1[t26] = matmul$1[t26] + float32((A[((A.stride.1*t34) - t33) + matmul$1.s1.k$x]*B[(B.stride.1*matmul$1.s1.k$x) + t35]))
+        const Expr wild_f32x = Variable::make(Float(32), "*");
+        vector<Expr> matches;
+        const Expr acc_pattern = wild_f32x + wild_f32x;
+        if (expr_match(acc_pattern, store->value, matches)) {
+            // matmul[t26]
+            const Load *matmul_load = matches[0].as<Load>();
+
+            // float32((A[((A.stride.1*t34) - t33) + matmul$1.s1.k$x]*B[(B.stride.1*matmul$1.s1.k$x) + t35]))
+            const Cast *cast = matches[1].as<Cast>();
+
+            if (matmul_load && cast) {
+                // Check if the load is loading from the same place where the store is storing
+                // i.e. if this is an update operation
+                if (matmul_load->name == store->name && equal(matmul_load->index, store->index)) {
+                    // Yes, this is an update operation, now let's check cast to see if it is a matmul that can be converted
+                    // to wmma intrinsic
+
+                    // float32((A[((A.stride.1*t34) - t33) + matmul$1.s1.k$x]*B[(B.stride.1*matmul$1.s1.k$x) + t35]))
+                    const Expr wild_f16x = Variable::make(Float(16), "*");
+                    const Expr mul_pattern = wild_f16x * wild_f16x;
```

Comment:
Why not match the add and mul at the same time? It seems like this would simplify this code a fair bit:

    pattern = wild_f32x + cast<float>(wild_f16x) * wild_f16x);

Also, are you sure this is right that the intermediate multiply result is f16 and not f32? i.e. `f32(wild_f16 * wild_f16x)` vs. `f32(wild_f16) * f32(wild_f16)`

Reply:
I simplified the matching logic in 7f9719c749625df8a27ca258fa455d982cd6ea31 based on your suggestion, for some reason I was sure that the multiplication was done in lower precision, but that's not the case.